nni/compression/pytorch/quantization_speedup/integrated_tensorrt.py

```diff
@@ -0,0 +1,233 @@
+        This function builds an engine from an onnx model.
+        """
+        # Attention that, builder should be set to 1 because of the implementation of allocate_buffer
+        builder.max_batch_size = 1
+        builder.max_workspace_size = common.GiB(1)
+
+        if extra_layer_bit == 32 and config is None:
+            pass
+        elif extra_layer_bit == 8 and config is None:
+            # entire model in 8bit mode
+            builder.int8_mode = True
+            builder.int8_calibrator = calib
+        else:
+            builder.int8_mode = True
+            builder.fp16_mode = True
+            builder.int8_calibrator = calib
+            builder.strict_type_constraints = strict_datatype
+
+        # Parse onnx model
+        with open(model_file, 'rb') as model:
+            if not parser.parse(model.read()):
+                print ('ERROR: Fail to parse the ONNX file.')
+                for error in range(parser.num_errors):
+                    print (parser.get_error(error))
+                return None
+
+        # This design may not be correct if output more than one
+        for i in range(network.num_layers):
+            if config is None:
+                break
+            layer = network.get_layer(i)
+            if layer.name in config:
+                w_bit = config[layer.name]['weight_bit']
+                a_bit = config[layer.name]['activation_bit']
+                layer.precision = Precision_Dict[w_bit]
+                layer.set_output_type(0, Precision_Dict[a_bit])
+        # Build engine and do int8 calibration.
+        engine = builder.build_cuda_engine(network)
+        return engine
+
+def build_engine_without_calib(model_file, config=None, extra_layer_bit=32, strict_datatype=False):
+    with trt.Builder(TRT_LOGGER) as builder, builder.create_network(common.EXPLICIT_BATCH) as network, trt.OnnxParser(network, TRT_LOGGER) as parser:
+        """
+        This function builds an engine from an onnx model.
+        """
+        # Attention that, builder should be set to 1 because of the implementation of allocate_buffer
+        builder.max_batch_size = 1
+        builder.max_workspace_size = common.GiB(1)
+
+        if extra_layer_bit == 32 and config is None:
+            pass
+        elif extra_layer_bit == 8 and config is None:
+            # entire model in 8bit mode
+            builder.int8_mode = True
+        else:
+            pass
+            builder.int8_mode = True
+            builder.fp16_mode = True
+            builder.strict_type_constraints = strict_datatype
+
+        # Parse onnx model
+        with open(model_file, 'rb') as model:
+            if not parser.parse(model.read()):
+                print ('ERROR: Fail to parse the ONNX file.')
+                for error in range(parser.num_errors):
+                    print (parser.get_error(error))
+                return None
+
+        input_tensor = network.get_input(0)
+        input_tensor.dynamic_range = (-100, 100)
```

Comment:
Why input dynamic range set to (-100, 100)?

Reply:
Range (-100, 100) is just for testing. It has been deleted in latest commit.